src/Package/TestApp/Help/RHostClientHelpTestApp.cs

```diff
@@ -5,36 +5,57 @@
 using System.Threading;
 using System.Threading.Tasks;
 using System.Windows.Forms;
+using Microsoft.Common.Core;
 using Microsoft.R.Components.Help;
 using Microsoft.R.Host.Client.Test.Script;
 using Microsoft.UnitTests.Core.Threading;
 
 namespace Microsoft.VisualStudio.R.Interactive.Test.Help {
     class RHostClientHelpTestApp : RHostClientTestApp {
-        IHelpVisualComponent _component;
+        private IHelpVisualComponent _component;
+        private TaskCompletionSource<bool> _tcs = new TaskCompletionSource<bool>();
```

Comment:
If you want to asynchronously wait for event to happen, you can define `EventTaskSource` for it (for example, `EventTaskSources.IRSession.Mutated`)